File: crates/delonix-runtime-bin/templates/node/src/routes/health.ts
import type { FastifyInstance } from "fastify";

/** Liveness/readiness probes (used by the Delonixfile HEALTHCHECK). */
export async function healthRoutes(app: FastifyInstance): Promise<void> {
  app.get("/live", async () => ({ status: "alive" }));
  app.get("/ready", async () => ({ status: "ready" }));
}

File: bins/delonix-runtime-bin/templates/python/src/__MODULE__/api/health.py
"""Liveness and readiness probes (used by the Delonixfile HEALTHCHECK)."""

from __future__ import annotations

from fastapi import APIRouter

router = APIRouter()


@router.get("/live")
def live() -> dict[str, str]:
    """The process is up."""
    return {"status": "alive"}


@router.get("/ready")
def ready() -> dict[str, str]:
    """The app is ready to serve (extend with DB/cache checks)."""
    return {"status": "ready"}
